examples/randomized-benchmark/two-qubit-rb/readme.md

```diff
@@ -0,0 +1,86 @@
+# Randomized Benchmarking for two-qubit gates using baking
+
+Randomized benchmarking (RB) is a procedure to generate an 
+average figure-of-merit characterizing the fidelity of 
+qubit operations. This procedure is useful as full process tomography 
+may be prohibitively costly in terms of the number of required operations.   
+RB utilizes operations from the Clifford group, which in the case of two qubits contains 11520 elements
+(for $$n$$ qubits $$
+\left|\mathcal{C}_{n}\right|=2^{n^{2}+2 n} \prod_{j=1}^{n}\left(4^{j}-1\right)
+$$).
+
+We propose to illustrate the use of baking tool to perform this protocol.
+
+## The class *RBTwoQubits*
+
+The pattern behind RB is always the same: 
+- Generate a random sequence composed of Cliffords
+- Append to the sequence one Clifford reverting the action of the random sequence, such that the identity circuit is played
+- Measure the survival probability
+- Repeat prior steps for varying number of Cliffords within the sequence
+
+In order to have an easy way to derive all Cliffords of the two qubit Clifford group, 
+we use the decomposition illustrated in the following reference: https://arxiv.org/pdf/1210.7011.pdf.
+This decomposition allows an easy derivation of the group from the Clifford group for one qubit, and uses three main two qubit gates
+(that shall be decomposed later into a native set of gates by the user depending on its own hardware):
+- CNOT
+- SWAP
+- iSWAP
+
+We introduce a specific class that allows an easy generation and execution of multiple random Clifford sequences based on 
+few generic inputs the user shall provide:
+- a  *QuantumMachinesManager* instance, which will be used to launch  the series of jobs necessary to play all synthezised sequences 
+- a *config* dictionary, used to open a Quantum Machine and to store the waveforms randomly generated
+- *N_Clifford*, the number of desired Clifford per sequence. Here an array of different values can be provided to retrieve different truncations
+- *N_sequences*: Number of random sequences to be generated for averaging
+- *two_qb_gate_macros*: Dictionary containing macros taking a baking object as input and using
+ baking play statements on necessary quantum elements to perform the three 2 qubit gates mentioned above ("CNOT", "SWAP" and "iSWAP" which are also keys of the dictionary).  The native two-qubit gate macro shall always contain a baking align statement before and after the playing statements
+- *single_qb_macros*: Optional dictionary in the case where a single qubit gate is not simply a QUA play statement on one quantum element, one can specify a dictionary of 
+a set of macros. Note that the dictionary shall then contain macros for all single qubit generators, i.e "I", "X", "Y", "X/2", "Y/2", "-X/2", "-Y/2".
+- *qubit_register*: Tuple containing names of two target qubits 
```

Comment:
in what format are those defined? We don't have a definition of a "qubit" in QUA

Reply:
I know, it is defined to yield access through the Resolver to the specific quantum elements. How do you call "q0" when calling resolve.q("q0", channel="X)?